src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -686,6 +662,22 @@ CascadiaSettings::CascadiaSettings(SettingsLoader&& loader)
 
     for (const auto& profile : loader.userSettings.profiles)
     {
+        // If a generator stops producing a certain profile (e.g. WSL or PowerShell were removed) we should also
+        // stop including the matching user's profile in _allProfiles (since they're disfunctional anyways).
+        // A profile requires a generated parent if it has a source (signaling that it was generated previously).
+        if (const auto source = profile->Source(); !source.empty())
+        {
+            static constexpr auto isGenerated = [](const auto& profile) {
+                return profile->Origin() == OriginTag::Generated;

```

Comment:
or Fragment!

Reply:
Right, because a fragment has a source...
It looks trivial but it's kinda hard to keep everything in your mind.